oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/InMemoryOAuth2AuthorizationService.java

```diff
@@ -64,7 +69,7 @@ public OAuth2Authorization findByToken(String token, @Nullable TokenType tokenTy
 	}
 
 	private boolean hasToken(OAuth2Authorization authorization, String token, TokenType tokenType) {
-		if (OAuth2AuthorizationAttributeNames.STATE.equals(tokenType.getValue())) {
+		if (tokenType != null && OAuth2AuthorizationAttributeNames.STATE.equals(tokenType.getValue())) {
```

Comment:
I know we have an issue raised for this potential NPE (https://github.com/spring-projects-experimental/spring-authorization-server/issues/170), but this was generating issues for some Introspection scenarios as well, si I worked on a quick patch.